dedupe test skipping in common_distributed and test_distributed

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38078 dedupe test skipping in common_distributed and test_distributed**

`common_distributed` and `test_distributed` have some error codes that overlap but are for different reasons, for example, code 75 in `test_distributed` is "no cuda available" but in common_distributed it is "need at least 2 CUDA devices".

This is an issue because the tests in `test_distributed` now use the utils in `common_distributed`, so we could get the wrong reason for skipping tests.

It is also the source of test failures in https://github.com/pytorch/pytorch/pull/37990.

This diff makes it so that the test skipping logic is deduped and put into `common_distributed.py`, where it can be reused and then imported into `test_distributed`

Differential Revision: [D21466768](https://our.internmc.facebook.com/intern/diff/D21466768/)

rohan-varma committed:
dedupe test skipping in common_distributed and test_distributed

`common_distributed` and `test_distributed` have some error codes that overlap but are for different reasons, for example, code 75 in `test_distributed` is "no cuda available" but in common_distributed it is "need at least 2 CUDA devices".

This is an issue because the tests in `test_distributed` now use the utils in `common_distributed`, so we could get the wrong reason for skipping tests.

It is also the source of test failures in https://github.com/pytorch/pytorch/pull/37990.

This diff makes it so that the test skipping logic is deduped and put into `common_distributed.py`, where it can be reused and then imported into `test_distributed`

Differential Revision: [D21466768](https://our.internmc.facebook.com/intern/diff/D21466768/)

[ghstack-poisoned]

rohan-varma committed:
Update on "dedupe test skipping in common_distributed and test_distributed"


`common_distributed` and `test_distributed` have some error codes that overlap but are for different reasons, for example, code 75 in `test_distributed` is "no cuda available" but in common_distributed it is "need at least 2 CUDA devices".

This is an issue because the tests in `test_distributed` now use the utils in `common_distributed`, so we could get the wrong reason for skipping tests.

It is also the source of test failures in https://github.com/pytorch/pytorch/pull/37990.

This diff makes it so that the test skipping logic is deduped and put into `common_distributed.py`, where it can be reused and then imported into `test_distributed`

Differential Revision: [D21466768](https://our.internmc.facebook.com/intern/diff/D21466768/)

[ghstack-poisoned]

mrshenli commented on torch/testing/_internal/common_distributed.py:
```diff
@@ -17,16 +18,56 @@
 }
 
+def skip_if_no_gpu(func):
+    """ Nccl multigpu tests requires at least 2 GPUS. Skip if this is not met"""
```
(this is prior to this PR)

tests requires -> tests require

mrshenli commented on torch/testing/_internal/common_distributed.py:
```diff
@@ -17,16 +18,56 @@
+    return wrapper
+
+
+def skip_if_no_cuda_distributed(func):
```
Looks like this is always used together with `skip_if_no_gpu`, so we can get rid of this as well?

mrshenli reviewed (approved):
Thanks! This is great!

rohan-varma committed:
Update on "dedupe test skipping in common_distributed and test_distributed"


`common_distributed` and `test_distributed` have some error codes that overlap but are for different reasons, for example, code 75 in `test_distributed` is "no cuda available" but in common_distributed it is "need at least 2 CUDA devices".

This is an issue because the tests in `test_distributed` now use the utils in `common_distributed`, so we could get the wrong reason for skipping tests.

It is also the source of test failures in https://github.com/pytorch/pytorch/pull/37990.

This diff makes it so that the test skipping logic is deduped and put into `common_distributed.py`, where it can be reused and then imported into `test_distributed`

Differential Revision: [D21466768](https://our.internmc.facebook.com/intern/diff/D21466768/)

[ghstack-poisoned]

rohan-varma committed:
Update on "dedupe test skipping in common_distributed and test_distributed"


`common_distributed` and `test_distributed` have some error codes that overlap but are for different reasons, for example, code 75 in `test_distributed` is "no cuda available" but in common_distributed it is "need at least 2 CUDA devices".

This is an issue because the tests in `test_distributed` now use the utils in `common_distributed`, so we could get the wrong reason for skipping tests.

It is also the source of test failures in https://github.com/pytorch/pytorch/pull/37990.

This diff makes it so that the test skipping logic is deduped and put into `common_distributed.py`, where it can be reused and then imported into `test_distributed`

Differential Revision: [D21466768](https://our.internmc.facebook.com/intern/diff/D21466768/)

[ghstack-poisoned]
